### PR TITLE
README-de.md: add German translation.

### PR DESCRIPTION
**Pull request by paulkogel**
@petehunt may we crosspost this translation on our German React community site http://reactjs.de?
Will credit you as the original author and link to the GH repo.

Would be awesome!

### CODE REVIEW

File: README-de.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+* [React selbst](#react-selbst-lernen)
+* [`npm`](#npm-lernen)
+* [JavaScript „Bundler“](#javascript-bundler-benutzen-lernen)
+* [ES6](#es6-lernen)
+* [Routing](#routing-lernen)
+* [Flux](#flux-lernen)
+
+**Du musst nicht alle Themenblöcke lernen um React produktiv einsetzen zu können.** Schau dir den nächsten Themenblock erst an, wenn du im aktuellen Probleme hast, die du mit den bestehenden Mitteln nicht lösen kannst.
+
+Des weiteren gibt es ein paar Themen, die in der React-Community viel diskutiert werden, die „bleeding edge“ sind. Die folgenden Themen sind interessant, aber schwer zu verstehen und viel weniger weitverbreitet als die eben genannten. **Für die meisten Anwendungen sind sie nicht notwendig.**
```
Comment (gbirke) [inline README-de.md]: too many adjectives and adverbs? I'd say "... aber schwer zu verstehen und weniger verbreitet als ..."

Comment (paulkogel) [inline README-de.md]: Thanks, fixed.

File: README-de.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+* Facebook veröffentlicht nur Open-Source Software, die es auch selbst nutzt. Deshalb sind alle Tools um diese Software nicht für Projekte ausgelegt, die kleiner als Facebook sind.
+* Es gibt viel Marketing, das sich als React Einführung verkleidet
+
+Dieses Dokument setzt voraus, dass der Leser bereits eine Webseite mit HTML, CSS und JavaScript geschrieben hat.
+
+## Warum auf mich hören?
+
+Es gibt jede Menge widersprüchliche Empfehlungen und Best Practices zu React da draußen - warum solltest du auf mich hören?
+
+Ich war Mitglied des ursprünglichen Teams bei Facebook, das React gebaut und als Open-Source veröffentlicht hat. Mittlerweile arbeite ich nicht mehr bei Facebook sondern bei einem kleinen Startup, wodurch ich auch die Perspektive von außerhalb Facebooks kenne.
```
Comment (cfleschhut) [inline README-de.md]: @paulkogel Read through it all, very nice! :+1: Just a few tiny issues:
```
Mittlerweile arbeite ich nicht mehr bei Facebook, sondern…
```

File: README-de.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+Wichtiger Hinweis: `CommonJS` benutzt die `require()`-Funktion, um Module zu importieren. Viele glauben, dass diese Funktion etwas mit dem `require.js` Projekt zu tun haben muss. Aus mehreren technischen Gründen würde ich euch nahelegen auf `require.js` zu verzichten. Es ist außerdem nicht sehr populär im React Ökosystem.
+
+## ES6 lernen
+
+Neben JSX, das ihr im React Tutorial gelernt habt, kann euch noch andere merkwürdige Syntax in React Beispielcode begegnen. Sie kommt aus ES6, was die neueste Version von JavaScript ist, die ihr vielleicht noch nicht kennt. Da sie noch sehr neu ist, wird sie von Browsern noch nicht unterstützt, aber euer Bundler kann sie mit der richtigen Konfiguration für euch übersetzen.
+
+Wenn ihr einfach nur mit React loslegen wollt **könnt ihr ES6 lernen überspringen**, oder es beiläufig lernen während ihr eure React Anwendungen baut.
+
+Wohlmöglich wird euch gesagt, dass ES6 Klassen jetzt der neue Standardweg sein sollen React-Komponenten zu schreiben. Das ist nicht wahr. Die meisten Leute (inklusive Facebook) benutzen weiterhin `React.createClass()`.
```
Comment (cfleschhut) [inline README-de.md]: `Womöglich wird euch gesagt…`

File: README-de.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+## Routing lernen
+
+„Single-page Applications“ sind gerade der letzte Schrei. Sie sind Webseiten die nur einmal geladen werden und wenn der Nutzer auf einen Link oder Button klickt sorgt das JavaScript das auf der Seite läuft dafür, dass die URL in der Adresszeile aktualisiert wird ohne dass die Seite refresht wird. Das verwalten der URL in der Adresszeile wird vom sogenannten **Router** übernommen.
```
Comment (cfleschhut) [inline README-de.md]:
```
…und wenn der Nutzer auf einen Link oder Button klickt, sorgt das JavaScript…
```
```
Das Verwalten der URL…
```

File: README-de.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+**Wenn ihr Flux braucht merkt ihr das von selbst. Wenn ihr zweifelt, ob ihr es braucht, dann braucht ihr es nicht.**
+
+Wenn ihr euch entschieden habt Flux zu nutzen: [Redux](http://redux.js.org/) ist die populärste und bestdokumentierte Flux-Bibliothek, die es gibt. Es gibt _viele_ Alternativen und ihr werdet versucht sein euch viele davon anzuschauen und zwischen ihnen abzuwägen. Mein Tipp ist, einfach die populärste zu nehmen.
+
+## Inline Styles lernen
+
+In der Zeit vor React haben viele Leute CSS Styles wiederverwendbar gemacht mithilfe komplizierter Stylesheets, die von CSS-Präprozessoren wie SASS erstellt wurden. Da es React einfach macht, wiederverwendbare Komponenten zu schreiben, kann euch das helfen weniger kompliziertes CSS zu schreiben. Viele in der Community, darunter auch ich, experimentieren mit verschiedenen Lösungen, die Stylesheets ganz abschaffen.
+
+Aus mehreren Gründen ist das ist eine ziemlich verrückte. Media Queries werden schwieriger und es gibt mögliche Performanceeinschränkungen. **Wenn ihr mitten im React-Einstieg steckt, stylet alles wie ihr es bisher immer getan habt.**
```
Comment (cfleschhut) [inline README-de.md]: Aus mehreren Gründen ist das ~~ist~~ eine ziemlich verrückte Idee.

File: README-de.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+Sobald ihr ein Gefühl dafür bekommen habt wie React funktioniert, könnt ihr euch nach Alternativen umschauen. Eine beliebte davon ist [BEM](https://en.bem.info). Ich empfehle, euer CSS immer weniger von einem CSS-Präprozessor bauen zu lassen, da React eine bessere Lösung bietet Styles wiederzuverwenden - durch wiederverwendbare Komponenten - und da euer JavaScript Bundler effizientere Stylesheets für euch generieren kann (hierzu habe ich [einen Vortrag bei der OSCON](https://www.youtube.com/watch?v=VkTCL6Nqm6Y) gehalten). Ansonsten funktioniert React, wie jede JavaScript Bibliothek, natürlich hervorragend mit CSS-Präprozessoren.
+
+Alternativ könnt ihr auch [CSS Modules](http://glenmaddern.com/articles/css-modules) nutzen, oder genauer [react-css-modules](https://github.com/gajus/react-css-modules). Mit CSS Modules schreibt ihr wie gewohnt CSS (oder SASs/LESS/Stylus) aber ihr könnt trotzdem eure CSS-Dateien wie React Inline Styles verwalten und zusammenstellen. Und ihr müsst euch dabei keine Sorgen darum machen eure Klassennamen mit Methodologien wie BEM einmalig zu halten - CSS Modules sorgen mit ihrem Modulsystem für euch hinter den Kulissen dafür.
```
Comment (cfleschhut) [inline README-de.md]: wie gewohnt CSS (oder **Sass**/LESS/Stylus)**,** aber

File: README-de.md [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+
+Serverseitiges Rendern erfordert noch viele Tools, um es richtig aufzusetzen. Da es deutlich React-Komponenten unterstützt, die geschrieben wurden, ohne an serverseitiges Rendern zu denken, solltet ihr zunächst mal eure App schreiben und euch dann um serverseitiges Rendern kümmern. Ihr werdet nicht all eure Komponenten umschrieben müssen, um es zu unterstützen.
+
+## Immutable.js lernen
+
+[Immutable.js](https://facebook.github.io/immutable-js/) bietet eine Menge von Datenstrukturen, die helfen können bestimmte Performanceprobleme von React Anwendungen zu lösen. Es ist eine großartige Bibliothek, die ihr wahrscheinlich in Zukunft viel in euren Apps verwenden werdet, aber sie ist vollkommen unnötig bis ihr einen Sinn für die Performanceauswirkungen gewonnen habt.
+
+## Relay, Falcor etc. lernen
+
+Sind Technologien, die euch helfen die Zahl der AJAX-Requests zu reduzieren. Sie sind noch sehr brandneu. Wenn ihr keine Probleme mit zu vielen AJAX-Requests habt braucht ihr Relay und Falcor nicht.
```
Comment (cfleschhut) [inline README-de.md]: Again, sorry for the nitpicking :smile: 
And thanks for the translation effort!

File: README-de.md (hunk 4 above)
Comment (paulkogel) [inline README-de.md]: JFYI github has this very handy diff markdown syntax:

```diff
- Das verwalten der URL
+ Das Verwalten der URL
```

File: README-de.md (hunk 7 above)
Comment (paulkogel) [inline README-de.md]: Fixed in https://github.com/paulkogel/react-howto/commit/37a2474ad17f36b16338736b5bad890120099fa7. Thanks for the good eye. Improvements like this always welcome - esp. regarding commas :)

File: README-de.md (hunk 4 above)
Comment (cfleschhut) [inline README-de.md]: @paulkogel Perfect, thx! Didn’t know this yet :)

**Pull request merged**